packages/oats/src/driver.ts

```diff
@@ -106,7 +107,7 @@ export function generateFile(opts?: GenerateFileOptions): types.Resolve {
     return {
       importAs: moduleName,
       importFrom: generatedFile,
-      name: refToTypeName(localName),
+      name: options.nameMapper?.(refToTypeName(localName), 'value') ?? refToTypeName(localName),
```

Comment:
how do you know that the context needing this is a 'value' and not a 'shape'?

Reply:
To my understanding, while resolving reference, the value class is preferably used. Given the mapping function, it would make sense to mapping it as `value`, IMHO. We don't probably use either `shape` or `reflection` here, do we?

Reply:
I think that we do need to resolve references to shapes also. and to reflection types for that matter. Both can be composed of references.

Reply:
How do use `shape` or `reflection` as ref 🤔 ?
Well, if that is the case, we could pass the kind in `GenerateFileOptions` so that resolver could know in which kind the name should be resolved. Does it make sense to you?